src/sentry/sentry_metrics/indexer/base.py

```diff
@@ -189,6 +252,99 @@ def __getitem__(self, org_id: int) -> Mapping[str, Optional[int]]:
         return self.results[org_id]
 
 
+class UseCaseResults:
+    def __init__(self) -> None:
+        self.results: MutableMapping[str, KeyResults] = defaultdict(lambda: KeyResults())
```

Comment:
This is fine as an intermediate step but I would sort of expect this new class would replace `KeyResults` eventually after we've stood up the new performance-aware indexer consumer. So, you wouldn't have a `UseCaseResults` that knows about `KeyResults` but rather `KeyResults` would just be use-case aware when the project is finished. Same thing for the other two classes.

Are we on the same page about that?

Reply:
Right now I would like the behavour of `KeyResults` and `UseCaseResults` to stay as consistent as possible before that happens, that's why the composition is used here. When `KeyResults` is no longer being used, we will port the logic to `UseCaseResults`.

Reply:
But I think maybe we should consider some intermediate container even if we are deprecating `KeyResults` because without it, it will `results` in `UseCaseResults` will look like
```python
self.results: MutableMapping[str, MutableMapping[int, MutableMapping[str, Optional[int]]]] = defaultdict(lambda: defaultdict(dict))
```
Which is pretty hard to make sense of